src/main/scala/com/lunatech/pointingpoker/actors/Room.scala

```diff
@@ -0,0 +1,116 @@
+
+import java.util.UUID
+
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
+import akka.actor.{ActorRef => UntypedRef}
+import com.lunatech.pointingpoker.websocket.WSMessage
+import com.lunatech.pointingpoker.websocket.WSMessage.MessageType
+
+object Room {
+
+  sealed trait Command
+  final case class Join(user: User)                                     extends Command
+  final case class Leave(userId: UUID)                                  extends Command
+  final case class Vote(userId: UUID, estimation: String)               extends Command
+  final case class ClearVotes(userId: UUID)                             extends Command
+  final case class ShowVotes(userId: UUID)                              extends Command
+  final case class EditIssue(userId: UUID, issue: String)               extends Command
+  private[actors] final case class GetData(replyTo: ActorRef[Response]) extends Command
+
+  sealed trait Response
+  final case class DataStatus(data: RoomData) extends Response
+
+  final case class User(id: UUID, name: String, voted: Boolean, estimation: String, ref: UntypedRef)
+
+  final case class RoomData(users: List[User], currentIssue: String, issueLastEditBy: Option[UUID])
+  object RoomData {
+    val empty: RoomData = RoomData(List.empty[User], "", Option.empty[UUID])
+  }
+
+  def apply(roomId: UUID): Behavior[Command] =
+    Behaviors.setup[Command] { _ =>
+      receiveBehaviour(roomId, RoomData.empty)
+    }
+
+  private[actors] def receiveBehaviour(roomId: UUID, data: RoomData): Behavior[Command] =
+    Behaviors.receive[Command] { (context, message) =>
+      message match {
+        case Join(user) =>
+          val newData = data.copy(users = user :: data.users)
+          setupNewUser(user, roomId, newData)
+          broadcast(WSMessage(MessageType.Join, roomId, user.id, user.name), newData.users, context)
+          receiveBehaviour(roomId, newData)
+        case Vote(userId, estimation) =>
+          val newUsers = data.users.map { u =>
+            if (userId == u.id) u.copy(voted = true, estimation = estimation)
+            else u
+          }
+          broadcast(WSMessage(MessageType.Vote, roomId, userId, estimation), newUsers, context)
+          receiveBehaviour(roomId, data.copy(users = newUsers))
+        case ClearVotes(userId) =>
+          val newUsers = data.users.map(_.copy(voted = false, estimation = ""))
+          broadcast(
+            WSMessage(MessageType.Clear, roomId, userId, WSMessage.NoExtra),
+            newUsers,
+            context
+          )
+          receiveBehaviour(roomId, data.copy(users = newUsers))
+        case ShowVotes(userId) =>
+          broadcast(
+            WSMessage(MessageType.Show, roomId, userId, WSMessage.NoExtra),
+            data.users,
+            context
+          )
+          Behaviors.same
+        case Leave(userId) =>
+          val newUsers = data.users.filter(_.id != userId)
```

Comment:
Super nitpicky, but whenever I see a filter and a != I always switch it to
```suggestion
          val newUsers = data.users.filterNot(_.id == userId)
```
Take it or leave it.

Reply:
Done